services/s3/src/test/java/software/amazon/awssdk/services/s3/internal/crossregion/S3CrossRegionAsyncClientRedirectTest.java

```diff
@@ -49,8 +49,10 @@ public void setup() {
     @Override
     protected void stubRedirectSuccessSuccess(Integer redirect) {
         when(mockDelegateAsyncClient.listObjects(any(ListObjectsRequest.class)))
-            .thenReturn(CompletableFutureUtils.failedFuture(new CompletionException(redirectException(redirect, CROSS_REGION.id(),
-                                                                                                      null, null))))
+            .thenReturn(CompletableFutureUtils.failedFuture(new CompletionException(redirectException(redirect,
```

Comment:
Can we add a test case where it's 400 with a different error code to verify no redirect? Or do we have an existing test somewhere covering that case?

Reply:
good point, adding it